### DISCUSSION TITLE
Open file using "find" (Unix command)

### DISCUSSION THREAD

**Discussion by pedz** [emacs-helm/helm#2553] [Q&A]
I've been looking at the help text and I see a lot of features but I can't find one that fits my particular need.

The file I'm looking for is not in a project nor a git repository and my laptop doesn't have the locate database built (and I'd prefer to not use it in this case).  I'm at `some/dir` and I know that a file matching `prefix*.ext` exists somewhere below the current directory.

The `some/dir/..else` feature is almost what I'm looking for except it uses locate and I'm not looking for a directory but a file.  Although it does spark the question, could that be generalized for this case?

I can do `find-name-dired` and then pick the file but that's no fun! :-)

Also (this should probably be a separate question), do you know about "Spotlight" on a Mac?  I think it is always available, up to date, and has all the power of locate plus much more.  It seems like that would be a really cool source for helm to tap into.

**Comment by thierryvolpiatto** [comment]
Perry Smith ***@***.***> writes:

> I've been looking at the help text and I see a lot of features but I can't find one that fits my particular need.
>
> The file I'm looking for is not in a project nor a git repository and
> my laptop doesn't have the locate database built (and I'd prefer to
> not use it in this case). I'm at some/dir and I know that a file
> matching prefix*.ext exists somewhere below the current directory.
>
> The some/dir/..else feature is almost what I'm looking for except it
> uses locate

It can use anything else IIRC, fd or find at least.

> and I'm not looking for a directory but a file.

The best tool for this is fd, it is bound to C-/ in helm-find-files (you
have of course to install the fd executable).

> Although it does spark the question, could that be generalized for
> this case?

helm-find-files is fully configurable, you can use something else than
locate e.g. fd or find.

> I can do find-name-dired and then pick the file but that's no fun! :-)

indeed :-)

> Also (this should probably be a separate question), do you know about
> "Spotlight" on a Mac? I think it is always available, up to date, and
> has all the power of locate plus much more. It seems like that would
> be a really cool source for helm to tap into.

There is a source for spotlight in helm-x-files.el, I never used it as I
never used a Mac, maybe it still works, have a look.

> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you are subscribed to this thread.*Message ID: ***@***.***>


-- 
Thierry
